x-pack/plugins/index_lifecycle_management/server/routes/api/nodes/register_details_route.js

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+
+
+import { callWithRequestFactory } from '../../../lib/call_with_request_factory';
+import { isEsErrorFactory } from '../../../lib/is_es_error_factory';
+import { wrapEsError, wrapUnknownError } from '../../../lib/error_wrappers';
+import { licensePreRoutingFactory } from'../../../lib/license_pre_routing_factory';
+
+function formatStats(stats, nodeAttrs) {
+  return Object.entries(stats.nodes).reduce((accum, [nodeId, stats]) => {
```

Comment:
I don't feel strongly about this, so I don't mind if you prefer to leave this as-is. But if we use `find` instead of `for..of`, we can create a variable which makes it clearer that we're looking for a single match.

```js
function formatStats(stats, nodeAttrs) {
  return Object.entries(stats.nodes).reduce((accum, [nodeId, nodeStats]) => {
    const { attributes = {} } = nodeStats;
    const matchingNodeAttrs = Object.entries(attributes).find(([key, val]) => (
      `${key}:${value}` === nodeAttrs
    ));

    if (matchingNodeAttrs) {
      accum.push({
        nodeId,
        nodeStats,
      });
    }

    return accum;
  }, []);
}
```

Reply:
Is `formatStats` an accurate name? It looks like this code is selecting stats's nodes which match the `nodeAttrs`. So maybe a name like `findMatchingNodes` would be more descriptive?